src/cli/command-processor.ts

```diff
@@ -0,0 +1,225 @@
+import {Worker} from 'worker_threads';
+import yargs from 'yargs/yargs';
+import * as readline from 'readline';
+
+/**
+ * A flag for the worker thread to indicate it's ready to recieve messages.
+ */
+export const READY_FLAG = 'READY';
+
+/**
+ * An interface to formalize the message being sent to the worker thread.
+ */
+export interface IntentMessage {
+  intentType: string;
+  requestId: string;
+}
+
+/**
+ * A class containing all parameters needed to process an Identify command.
+ */
+export class IdentifyMessage implements IntentMessage {
+  intentType = 'IDENTIFY';
+  requestId: string;
+  discoveryBuffer: string;
+  deviceId: string;
+
+  /**
+   * @param requestId  The request id for a triggered Identify request.
+   * @param discoveryBuffer  The discovery buffer to send in a triggered Identify request.
+   * @param deviceId  The device id for a triggered Identify request.
+   * @returns  A new IntentMessage instance.
+   */
+  constructor(requestId: string, discoveryBuffer: string, deviceId: string) {
+    this.requestId = requestId;
+    this.discoveryBuffer = discoveryBuffer;
+    this.deviceId = deviceId;
+  }
+}
+
+/**
+ * A class containing all parameters needed to process an Identify command.
+ */
+export class ExecuteMessage implements IntentMessage {
```

Comment:
ccould we use the existing local home sdk types?

Reply:
These are for formatting messages to pass to the worker, so they're different from the ones in the Local Home SDK.